docs/src/modules/components/MarkdownDocs.js

```diff
@@ -37,81 +38,104 @@ const styles = theme => ({
 const demoRegexp = /^"demo": "(.*)"/;
 const SOURCE_CODE_ROOT_URL = 'https://github.com/mui-org/material-ui/tree/master';
 
-function MarkdownDocs(props, context) {
-  const { classes, demos, disableAd, markdown, markdownLocation: markdownLocationProp } = props;
-  const contents = getContents(markdown);
-  const headers = getHeaders(markdown);
+class MarkdownDocs extends React.Component {
+  state = {
+    userLanguage: 'en',
+  };
 
-  let markdownLocation = markdownLocationProp || context.activePage.pathname;
-
-  if (!markdownLocationProp) {
-    const token = markdownLocation.split('/');
-    token.push(token[token.length - 1]);
-    markdownLocation = token.join('/');
+  componentDidMount() {
+    this.setState({ userLanguage: acceptLanguage.get(navigator.language) || 'en' });
```

Comment:
Don't we duplicate this logic? Why not using redux or the context to propagate it?